tests/formats/psf-molfile.cpp

```diff
@@ -0,0 +1,84 @@
+// Chemfiles, a modern library for chemistry file reading and writing
+// Copyright (C) Guillaume Fraux and contributors -- BSD license
+
+#include "catch.hpp"
+#include "helpers.hpp"
+#include "chemfiles.hpp"
+using namespace chemfiles;
+
+TEST_CASE("Read PSF format") {
+    SECTION("Standard PSF") {
+        auto file = Trajectory("data/psf/776wat_1Ca.psf");
+        CHECK(file.nsteps() == 1);
+        auto frame = file.read();
+
+        CHECK(frame.size() == 3105);
+
+        auto& topology = frame.topology();
+        CHECK(topology[0].name() == "OH2");
+        CHECK(topology[0].type() == "OT");
+        CHECK(topology[0].charge() == 0.0);
+        CHECK(topology[0].mass() == static_cast<double>(15.9994f));
+
+        CHECK(topology[1].name() == "H1");
+        CHECK(topology[1].type() == "HT");
+        CHECK(topology[1].charge() == static_cast<double>(0.556400f));
+        CHECK(topology[1].mass() == static_cast<double>(1.0080f));
+
+        CHECK(topology[3104].name() == "CAL");
+        CHECK(topology[3104].type() == "CAL");
+        CHECK(topology[3104].charge() == 2.0);
+        CHECK(topology[3104].mass() == static_cast<double>(40.0800f));
+
+        // FIXME: the molfile plugin does not read residue information, we
+        // should add it when re-implementing a PSF reader.
```

Comment:
Surely I missed something, but I think molfiles does read residues. Wouldn't the following three lines fix this?
The resids in the water test file don't seem sensible, but overall this works principally.
```patch
diff --git a/src/formats/Molfile.cpp b/src/formats/Molfile.cpp
index cd141548..0a81452b 100644
--- a/src/formats/Molfile.cpp
+++ b/src/formats/Molfile.cpp
@@ -298,6 +298,9 @@ template <MolfileFormat F> void Molfile<F>::read_topology() {
         }
         atom_id++;
     }
+    for (auto res : residues) {
+        topology_->add_residue(std::move(res.second));
+    }
 
     if (plugin_handle_->read_bonds == nullptr) {
         return;
```

Reply:
Right, we are reading them but not adding them to the topology ... Good catch!!